lib/run_part.c

```diff
@@ -61,7 +61,7 @@ int run_parts (const char *directory, const char *name, const char *action)
 		struct stat  sb;
 
 		if (asprintf(&s, "%s/%s", directory, namelist[n]->d_name) == -1) {
-			fprintf(stderr, "could not allocate memory\n");
+			fprintf(shadow_logfd, "could not allocate memory\n");
```

Comment:
There are a lot of perror(3) calls in nearby code.  Why use `shadow_logfd` here and stderr (implied by perror(3)) elsewhere?  I see there 's one fprintf() below, but perror(3) seems to be the most common one.

I would maybe replace this by `perror("asprintf");`